options/posix/include/nl_types.h

```diff
@@ -1,6 +1,22 @@
 #ifndef NL_TYPES_H
 #define NL_TYPES_H
 
+#ifdef __cplusplus
+extern "C" {
+#endif
 
+#define NL_SETD 1
+#define NL_CAT_LOCALE 1
 
-#endif // NL_TYPES_H 
\ No newline at end of file
+typedef int nl_item;
+typedef void *nl_catd;
```

Comment:
I'd rather not make this a void pointer, we should add a struct and make this a pointer to the struct. Since this interface uses opaque types we can then easily change the internal struct later.

Reply:
Oh I forgot to limit changes, Im doing some local changes. Sorry about that.

Reply:
Lemme open another pull request.